test/state/player_state_test.cpp

```diff
@@ -84,6 +84,8 @@ TEST_F(PlayerStateTest, GetBotByIdTest) {
     EXPECT_EQ(bot1, getBotById(player_states[0], 2000));
     EXPECT_EQ(bot2, getBotById(player_states[1], 65));
     EXPECT_EQ(Bot::null, getBotById(player_states[0], -50));
+    EXPECT_NE(Bot::null, getBotById(player_states[0], 2000));
```

Comment:
```PlayerStateTest``` isn't the place for operator overload feature test. These are logical tests pertaining to the use of player_state. 
In this case, the ```EXPECT_NE``` statements are redundant, as they achieve the same logic test as the previous ```EXPECT_EQ``` statement.
[bot_test.cpp](https://github.com/delta/codecharacter-simulator-2020/blob/master/test/state/bot_test.cpp) and [tower_test.cpp](https://github.com/delta/codecharacter-simulator-2020/blob/master/test/state/tower_test.cpp) are the correct places to add this test. See [vector equality tests](https://github.com/delta/codecharacter-simulator-2020/blob/5af3fa6b3fb76a1f5324de71ac41f1299ade70d6/test/physics/vector_test.cpp#L9) for correct usage.